### PR TITLE
Add response status code test

### PR DESCRIPTION
**Pull request by alex-james-dev**
- Created a new test group `response status line` to test that `package:http` clients correctly process the response status line.

- Added a new test `without status code` to the `response status line` test group.

- The `without status code` test checks that `package:http` clients can handle an HTTP response status line with a missing status code, e.g. `HTTP/1.1 OK`.

- `cronet_http` is failing `without status code test`, we are skipping the test until we fix this issue: #1013.

- Created a new test server `response_status_line_server.dart` that receives a status line from the client and sets this as the response status line.

---

- [x] I’ve reviewed the contributor guide and applied the relevant portions to this PR.

<details>
  <summary>Contribution guidelines:</summary><br>

- See our [contributor guide](https://github.com/dart-lang/.github/blob/main/CONTRIBUTING.md) for general expectations for PRs.
- Larger or significant changes should be discussed in an issue before creating a PR.
- Contributions to our repos should follow the [Dart style guide](https://dart.dev/guides/language/effective-dart) and use `dart format`.
- Most changes should add an entry to the changelog and may need to [rev the pubspec package version](https://github.com/dart-lang/sdk/wiki/External-Package-Maintenance#making-a-change).
- Changes to packages require [corresponding tests](https://github.com/dart-lang/.github/blob/main/CONTRIBUTING.md#Testing).

Note that many Dart repos have a weekly cadence for reviewing PRs - please allow for some latency before initial review feedback.
</details>

### CODE REVIEW

File: pkgs/http_client_conformance_tests/lib/src/response_status_line_server.dart [hunk 1]
```diff
@@ -0,0 +1,36 @@
+// Copyright (c) 2022, the Dart project authors.  Please see the AUTHORS file
```
Comment (brianquinlan) [inline pkgs/http_client_conformance_tests/lib/src/response_status_line_server.dart:1]: 2023

File: pkgs/http_client_conformance_tests/lib/src/response_status_line_server.dart [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:io';
+
+import 'package:stream_channel/stream_channel.dart';
+
+/// Starts an HTTP server that returns custom headers.
```
Comment (brianquinlan) [inline pkgs/http_client_conformance_tests/lib/src/response_status_line_server.dart:10]: returns a custom status line.

File: pkgs/http_client_conformance_tests/lib/src/response_status_line_server.dart [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+///
+/// Channel protocol:
+///    On Startup:
+///     - send port
+///    On Request Received:
+///     - load response header map from channel
```
Comment (brianquinlan) [inline pkgs/http_client_conformance_tests/lib/src/response_status_line_server.dart:16]: Remove this line

File: pkgs/http_client_conformance_tests/lib/src/response_status_line_tests.dart [hunk 4]
```diff
@@ -0,0 +1,33 @@
+// Copyright (c) 2022, the Dart project authors.  Please see the AUTHORS file
```
Comment (brianquinlan) [inline pkgs/http_client_conformance_tests/lib/src/response_status_line_tests.dart:1]: 2023

**Review by brianquinlan — COMMENTED** [review]
Please update the PR message.

**Comment by brianquinlan** [comment]
cronet_http fails this test - maybe mark is as skipped with a reason like 'currently failing in cronet_http, see XXX` where XXX is a link to an issue that you file against cronet_http ;-)

File: pkgs/http_client_conformance_tests/lib/src/response_status_line_server.dart (hunk 1 above)
Comment (alex-james-dev) [reply pkgs/http_client_conformance_tests/lib/src/response_status_line_server.dart]: :heavy_check_mark: Done!

File: pkgs/http_client_conformance_tests/lib/src/response_status_line_server.dart (hunk 2 above)
Comment (alex-james-dev) [reply pkgs/http_client_conformance_tests/lib/src/response_status_line_server.dart]: :heavy_check_mark: Done!

File: pkgs/http_client_conformance_tests/lib/src/response_status_line_server.dart (hunk 3 above)
Comment (alex-james-dev) [reply pkgs/http_client_conformance_tests/lib/src/response_status_line_server.dart]: :heavy_check_mark: Done!

File: pkgs/http_client_conformance_tests/lib/src/response_status_line_tests.dart (hunk 4 above)
Comment (alex-james-dev) [reply pkgs/http_client_conformance_tests/lib/src/response_status_line_tests.dart]: :heavy_check_mark: Done!

**Comment by alex-james-dev** [comment]
Hey @brianquinlan,

- The PR message has been updated.

- An issue has been filed against `cronet_http` because it fails the `without status code` test in the `response status line` test group.

- Until we fix this issue, we will skip the `without status code` test and output the following message instead:

```
Enable after https://github.com/dart-lang/sdk/issues/ is fixed
```

This PR is now ready for review! :tada:

**Review by brianquinlan — APPROVED** [review]

**Pull request merged**